packages/core/src/nodes.rs

```diff
@@ -894,3 +897,70 @@ impl<'a, T: IntoAttributeValue<'a>> IntoAttributeValue<'a> for Option<T> {
         }
     }
 }
+
+#[derive(Debug)]
+pub enum KeyValue<'a> {
+    /// Text attribute key value
+    Text(&'a str),
+
+    /// Signed integer key value
+    Int(i64),
+}
+
+// need to understand if these traits are necessary
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serialize", serde(untagged))]
+pub enum BorrowedKeyValue<'a> {
```

Comment:
This was necessary to support Arbitrary Any values in Arbitrary Attribute but for Keys Arbitrary values make less sense so we don't need a borrowed variant